web3.js/src/address-lookup-table-program.ts

```diff
@@ -0,0 +1,295 @@
+import {toBufferLE} from 'bigint-buffer';
+import * as BufferLayout from '@solana/buffer-layout';
+
+import * as Layout from './layout';
+import {PublicKey} from './publickey';
+import * as bigintLayout from './util/bigint';
+import {SystemProgram} from './system-program';
+import {TransactionInstruction} from './transaction';
+import {encodeData, IInstructionInputData} from './instruction';
+
+export type CreateLookupTableParams = {
+  /** Account used to derive and control the new address lookup table. */
+  authority: PublicKey;
+  /** Account that will fund the new address lookup table. */
+  payer: PublicKey;
+  /** A recent slot must be used in the derivation path for each initialized table. */
+  recentSlot: bigint;
+};
+
+export type FreezeLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+};
+
+export type ExtendLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+  /** Account that will fund the table reallocation. */
```

Comment:
Can you also say that the payer isn't required if the reallocation has been funded already?

Reply:
Fixed.

Reply:
Fixed.